src/Microsoft.QualityGuidelines.Analyzers/Core/DoNotRaiseExceptionsInExceptionClauses.cs

```diff
@@ -0,0 +1,95 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftQualityGuidelinesAnalyzersResources.DoNotRaiseExceptionsInExceptionClausesDescription), MicrosoftQualityGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftQualityGuidelinesAnalyzersResources));
+
+        private const string helpLinkUrl = "https://msdn.microsoft.com/en-us/library/bb386041.aspx";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: helpLinkUrl,
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterOperationBlockAction(operationBlockContext =>
+            {
+                foreach (var block in operationBlockContext.OperationBlocks)
+                {
+                    var walker = new ThrowInsideFinallyWalker();
+                    walker.Visit(block);
+
+                    foreach (var throwStatement in walker.ThrowStatements)
+                    {
+                        operationBlockContext.ReportDiagnostic(throwStatement.Syntax.CreateDiagnostic(Rule));
+                    }
+                }
+            });
+        }
+
+        /// <summary>
+        /// Walks a IOperation tree to find throw statements inside finally blocks.
+        /// </summary>
+        private class ThrowInsideFinallyWalker : OperationWalker
+        {
+            private int _finallyBlockNestingDepth;
+
+            public List<IThrowStatement> ThrowStatements { get; private set; } = new List<IThrowStatement>();
+
+            public override void VisitTryStatement(ITryStatement operation)
+            {
+                Visit(operation.Body);
+                foreach (var catchClause in operation.Catches)
+                {
+                    Visit(catchClause);
+                }
+
+                _finallyBlockNestingDepth++;
+                Visit(operation.FinallyHandler);
+                _finallyBlockNestingDepth--;
+            }
```

Comment:
No because this method does what base.VisitTryStatement does with the finally tracking added. Calling base.VisitTryStatemetn would call the nodes to be visited again.